src/pydot/core.py

```diff
@@ -288,19 +288,52 @@ def __repr__(self):
         return f"frozendict({dict_repr})"
 
 
+def make_quoted(s):
+    """Transform a string into a quoted string, escaping specials."""
+    replace = {
+        '"': r"\"",
+        "\n": r"\n",
+        "\r": r"\r",
+    }
+    s.translate(replace)
```

Comment:
Looks like the string is translated twice here, once through `s.translate()` (but its result isn't assigned to anything), and once through the `for` loop.

Reply:
You're right, I don't know what I was thinking when I wrote this.

There's no need for the loop at all, it should just `return rf'"{s.translate(replace)}"'` at the end.

Reply:
Additionally, I appear to have completely forgotten that `str.translate()` takes a mapping of _ordinals_ to ordinals-or-strings. So the correct mapping is:

```python
replace = {
    ord('"'): r"\"",
    ord("\n"): r"\n",
    ord("\r"): r"\r",
}
```

Fixed now.